folly/Synchronized.h

```diff
@@ -135,6 +135,7 @@ class SynchronizedBase<Subclass, true> {
    *     return data.getValue();
    *   });
    */
+#if __cplusplus > 201103L
```

Comment:
Folly code uses the subset of C++14 that is supported by GCC 4.8.

This particular case can be tweaked to build with C++11 by adding a return type before the opening brace, using either `decltype` or `std::result_of`. Both would work.